docs/customization.rst

```diff
@@ -175,100 +198,287 @@ Configuration Options
+    /2.0
 
-        A dictionary used to populate the version dropdown.  If this variable is provided, the static
-        dropdown is used and any JavaScript information is ignored.
+Whereas Sphinx must be executed seperately for each version of the documentation you are
+building.
+
+.. code-block:: text
+
+    git fetch --all --tags
+    git checkout 1.0
+    sphinx-build docs _build/1.0
+    git checkout 2.0
+    sphinx-build docs _build/2.0
+
+This means you need to implement a custom mechanism that allows keeping the built
+documentation when checking out different branches or tagged commits.
+
+.. important::
+    To use the version dropdown selector, you must set :confval:`version_dropdown` to `True` in
+    the `html_theme_options` dict.
+
+    .. code-block:: python
+
+        html_theme_options = {
+            "version_dropdown": True,
+        }
+
+Supported Approaches
+********************
+
+There are two approaches:
+
+1. The version information is stored in a :confval:`version_info` `list` in the conf.py file.
+
+   .. note::
+       Notice this is an ordered list. Meaning, approach 1 will take precedence over approach 2.
+
+2. The version information is stored in a JSON file.
+
+   The default name of the JSON file is ``versions.json``, but the JSON file's name could be
+   changed by setting :confval:`version_json` in the conf.py file's `html_theme_options`.
+
+   .. code-block:: python
+
+       html_theme_options = {
+           "version_json": "doc_versions.json",
+       }
+
+   .. warning::
+       The JSON approach only works if your documentation is served from a webserver; it does not
+       work if you use ``file://`` url). When serving the docs from a webserver the
+       :confval:`version_json` file is resolved relative to the *parent* directory that contains
+       the sphinx builder's HTML output. For example, if the builder's output is ``2.0``, you
+       should have directory structure like so:
+
+       .. code-block:: text
+
+           /versions.json
+           /1.0/index.html
+           /2.0/index.html
+
+
+Version Information Structure
+*****************************
+
+Both approaches use a data structure similar to what is used by the
+`mkdocs mike plugin <https://github.com/jimporter/mike>`_. Contrary to what the mike plugin's
+README says, the ``aliases`` field is not optional, but it can be set to an empty list if not using
+aliases. Other required fields include ``version`` and ``title``.
+
+- The ``version`` field can be set to a relative/absolute path or a URL.
+- The ``title`` field is a string used to describe the version in the selector's dropdown menu.
+- The ``aliases`` field is meant for giving a specific version a surname like "latest" or "stable".
+  This way the alias specified will redirect to the corresponding ``version``'s path. 
+
+  .. note::
+      Aliases are not specified via the dropdown menu. Aliases are used for command line arguments
+      to the mike plugin's CLI.
```

Comment:
Should I re-word this to mention how using "URL/stable" will redirect to "URL/2.0"?

Do I even need to mention the mike plugin's CLI at all?

Reply:
I don't think we should mention the mike plugin cli since I don't think it applies to sphinx, we are just using the same json format.

Reply:
I took this note out, re-worded the initial brief explanation of `aliases`, and added a note after the example snippet from these docs that use external URLs